client/src/App.js

```diff
@@ -3,31 +3,143 @@ import { Switch, Route } from 'react-router-dom';
 import { saveToLocalStorage, loadFromLocalStorage } from './lib/localStorage';
 import styled from 'styled-components/macro';
 import Home from './pages/Home.js';
-import BurgerMenu from './BurgerMenu.js';
+import BurgerMenu from './components/BurgerMenu.js';
 import MarketCard from './pages/MarktCard.js';
 import MarketForm from './pages/MarktForm.js';
 
 function App() {
   const [markets, setMarkets] = useState(loadFromLocalStorage('Markets') ?? []);
+  const [bookmarkedMarkets, setBookmarkedMarkets] = useState(
+    loadFromLocalStorage('bookmarkedMarkets') ?? []
+  );
+  const [filteredMarkets, setFilteredMarkets] = useState([]);
 
   function addComment(comment, marketToUpdate) {
-    const updatedMarkets = markets.map((market) => {
-      // @TODO: Replace .name with ._id later when markets are fetched from API / DB
-      if (market.name === marketToUpdate.name) {
+    const upToDateMarkets = markets.filter(
+      (market) => market._id !== marketToUpdate._id
+    );
+    markets.map((market) => {
+      if (market._id === marketToUpdate._id) {
         market.comments.push(comment);
+        fetch('http://localhost:4000/market/' + marketToUpdate._id, {
+          method: 'PUT',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(marketToUpdate),
+        })
+          .then((result) => result.json())
+          .then((updatedMarket) => {
+            setMarkets([...upToDateMarkets, updatedMarket]);
+          })
+          .catch((error) => console.error(error));
       }
       return market;
     });
+  }
 
-    setMarkets(updatedMarkets);
+  function addRating(rating, marketToUpdate) {
```

Comment:
Your functions `addComment` and `addRating` are very similar. The only distinction being, that the one updates the ratings and the other the comments, but the rest of the code is almost identical. 
Keeping the DRY principles in mind, this calls for refactoring 😄 Maybe you find a way to improve this to have one function that handles both use cases. Not something that needs to be done in this PR but keep that as possible improvement for a future sprint.